upstash/redis/database/resource.go

```diff
@@ -0,0 +1,66 @@
+package database
+
+import (
+	"github.com/hashicorp/terraform-plugin-sdk/v2/helper/schema"
+)
+
+func ResourceDatabase() *schema.Resource {
+	return &schema.Resource{
+		CreateContext: resourceDatabaseCreate,
+		ReadContext:   resourceDatabaseRead,
+		UpdateContext: resourceDatabaseUpdate,
+		DeleteContext: resourceDatabaseDelete,
+		Schema: map[string]*schema.Schema{
+			"database_id": &schema.Schema{
+				Type:        schema.TypeString,
+				Computed:    true,
+				Description: "Unique Database ID for created database",
+			},
+			"database_name": &schema.Schema{
+				Type:        schema.TypeString,
+				Required:    true,
+				ForceNew:    true,
+				Description: "Name of the database",
+			},
+			"region": &schema.Schema{
+				Type:     schema.TypeString,
+				Required: true,
+				ForceNew: true,
+				Description: "region of the database. Possible values are: " +
+					"\"eu-west-1\", \"us-east-1\", \"us-west-1\", \"ap-northeast-1\" , \"eu-central1\"",
+			},
+			"endpoint": &schema.Schema{
+				Type:        schema.TypeString,
+				Computed:    true,
+				Description: "Database URL for connection",
+			},
+			"password": &schema.Schema{
+				Type:        schema.TypeString,
+				Computed:    true,
+				Sensitive:   true,
+				Description: "Password of the database",
+			},
+			"consistent": &schema.Schema{
+				Type:        schema.TypeBool,
+				Optional:    true,
+				Default:     false,
+				ForceNew:    true,
+				Description: "When enabled database runs in Consistency Mode",
```

Comment:
We can write a better description here sth like "When enabled all writes are synchronously persisted to the disk"